resources/js/common/components/SEO/SEO.tsx

```diff
@@ -0,0 +1,251 @@
+   */
+  twitterDescription?: string;
+
+  /**
+   * Image to display in Twitter shares.
+   * @default Same as `ogImage`.
+   */
+  twitterImage?: string;
+
+  /**
+   * Twitter handle of our website account.
+   * @default "@RetroCheevos"
+   */
+  twitterSite?: string;
+
+  // Article specific metadata (only used if `type` is "article").
+  /**
+   * When the article was first published.
+   * Must be ISO-8601 format.
+   * @example "2024-01-01T00:00:00Z"
+   */
+  publishedTime?: string;
+
+  /**
+   * When the article was last modified.
+   * Must be ISO-8601 format.
+   * @example "2024-01-01T00:00:00Z"
+   */
+  modifiedTime?: string;
+
+  /**
+   * Article author displayNames. Used for article schema and Open Graph author tags.
+   */
+  authors?: string[];
+
+  /**
+   * Article section/category.
+   * @example "Game Guides", "News"
+   */
+  section?: string;
+
+  /**
+   * Article tags/keywords.
+   */
+  tags?: string[];
+
+  /**
+   * Custom structured data (JSON-LD).
+   * Greatly improves Google SEO juice.
+   * @see https://developers.google.com/search/docs/appearance/structured-data/intro-structured-data
+   */
+  jsonLd?: Record<string, unknown>;
+}
+
+/**
+ * SEO component for managing page metadata and improving social media sharing embeds.
+ * It handles Open Graph tags (for Facebook/Discord), Twitter cards, and structured data.
+ *
+ * @example
+ * // Basic usage
+ * <SEO
+ *   title="Sonic the Hedgehog"
+ *   description="There are 36 achievements worth 305 points. Sonic the Hedgehog for Genesis/Mega Drive - explore and compete on this classic game at RetroAchievements."
+ * />
+ */
+export const SEO: FC<SEOProps> = ({
+  title,
+  description,
+  canonical,
+  type = 'website',
+
+  // OG defaults to base metadata if not provided.
+  ogTitle = title,
+  ogDescription = description,
+  ogImage = '/assets/images/favicon.webp',
+  ogSiteName = 'RetroAchievements',
+
+  // Twitter defaults to OG if not provided.
+  twitterCard = 'summary_large_image',
+  twitterTitle = ogTitle,
+  twitterDescription = ogDescription,
+  twitterImage = ogImage,
+  twitterSite = '@RetroCheevos',
+
+  // Article metadata.
+  /** ISO-8601 timestamp */
+  publishedTime,
+  /** ISO-8601 timestamp */
+  modifiedTime,
+  authors,
+  section,
+  tags,
+
+  // Optional structured data.
+  jsonLd,
+}) => {
+  const { constructedCanonicalUrl } = useConstructedCanonicalUrl();
+
+  const resolvedCanonicalUrl = canonical ?? constructedCanonicalUrl;
+
+  const defaultJsonLd = {
+    '@context': 'https://schema.org',
+    '@type': 'WebSite',
+    name: title,
+    description,
+    url: resolvedCanonicalUrl,
+  };
+
+  const isArticle = type === 'article';
+
+  return (
+    <Head title={title}>
+      {/* Base metadata */}
+      <meta name="description" content={description} />
+      <link rel="canonical" href={resolvedCanonicalUrl} />
+
+      {/* Open Graph metadata */}
+      <meta property="og:site_name" content={ogSiteName} />
```

Comment:
Are these all supposed to end with `inertia`?:
```
<meta property="og:site_name" content="RetroAchievements" inertia>
```

Reply:
Yea - Inertia does this on purpose as a runtime performance optimization.

On client-side navigation, Inertia needs to quickly identify which head elements it inserted during potential previous navigations, as well as determine new ones to add (if any). The `inertia` attribute helps Inertia quickly access those tags with a selector.